### PR TITLE
Proposed changes to "Unit tests CMake and Travis refactoring"

### PR DESCRIPTION
**Pull request by maflcko**
I think it makes sense to run the tests with gcc and clang.

### CODE REVIEW

**Comment by maflcko** [comment]
Feel free to squash instead of a pull.

File: .travis.yml [hunk 1]
```diff
@@ -34,8 +34,10 @@ matrix:
   include:
     - compiler: gcc
       env: TASK="test-core"
-    - env: TASK="test-extended"
-    - env: TASK="doc"
+    - compiler: clang
```
Comment (SergioRAgostinho) [inline .travis.yml]: Setting the compiler to clang is causing the tests all tests to segfault when they run. I have a gut feeling it's an issue with our dependencies being compiled with g++ and probably a different version of the std c++ lib. So I would suggest to leave this one as well to gcc.

**Review by SergioRAgostinho — CHANGES_REQUESTED** [review]
Thanks for the fixes.

File: .travis.yml (hunk 1 above)
Comment (maflcko) [reply .travis.yml]: I think your gut feeling is right. All green now.

**Review by SergioRAgostinho — APPROVED** [review]

**Pull request merged**